src/components/ip-checkbox-list/ip-checkbox-list.tsx

```diff
@@ -0,0 +1,67 @@
+import { Component, Prop, h, State, Watch, EventEmitter, Event } from '@stencil/core';
+
+interface CheckboxOption {
+  id: string;
+  label: string;
+}
+
+@Component({
+  tag: 'ip-checkbox-list',
+  styleUrl: 'ip-checkbox-list.scss',
+  shadow: true,
+})
+export class IpCheckboxList {
+  @Prop() options: string;
+  @State() selectedOptions: string[] = [];
+
+  parsedOptions: CheckboxOption[] = [];
+
+  @Event({ bubbles: true, composed: true }) selectionChanged: EventEmitter<string[]>;
+
+  componentWillLoad() {
+    this.parseOptions(this.options);
+  }
+
+  @Watch('options')
+  parseOptions(newValue: string) {
+    try {
+      this.parsedOptions = JSON.parse(newValue);
+    } catch (error) {
+      console.error('Invalid options:', error);
+    }
```

Comment:
that very dangerous. you must validate that your received JSON fit with your expected structure.